docs/cancel-and-close-unpaid-orders-after-two-days/script.liquid

```diff
@@ -56,143 +28,266 @@
+    {% if order.cancelledAt == blank %}
+      {% assign summary["qualifies_for_cancellation"] = true %}
+    {% else %}
+      {% assign summary["qualifies_for_cancellation"] = false %}
+    {% endif %}
+
+    {% if order.closedAt == blank %}
+      {% assign summary["qualifies_for_closing"] = true %}
+    {% else %}
+      {% assign summary["qualifies_for_closing"] = false %}
+    {% endif %}
+
+    {% assign summary["qualifies_for_tagging"] = false %}
+    {% if options.tag_to_add_to_the_order != blank %}
+      {% assign order_tags = order.tags | downcase %}
+      {% assign order_tag_to_match = options.tag_to_add_to_the_order | downcase | strip %}
+      {% unless order_tags contains order_tag_to_match %}
+        {% assign summary["qualifies_for_tagging"] = true %}
+      {% endunless %}
+    {% endif %}
+
+    {% unless summary.qualifies_for_cancellation or summary.qualifies_for_closing or summary.qualifies_for_tagging %}
+      {% continue %}
+    {% endunless %}
+
+    {% if options.test_mode__boolean %}
+      {% assign test_mode_summaries[test_mode_summaries.size] = summary %}
+      {% continue %}
+    {% endif %}
+
+    {% log summary %}
+
+    {% if summary.qualifies_for_cancellation %}
+      {% action "shopify" %}
+        mutation {
+          refundCreate(input: {
+            orderId: {{ order.id | json }}
             {% assign transaction = order.transactions.first %}
             {% if options.void_payment_when_possible__boolean and transaction != nil %}
-              "transactions": [
+              transactions: [
                 {
-                  "parent_id": {{ transaction.id | json }},
-                  "amount": "0.0",
-                  "kind": "void",
-                  "gateway": {{ transaction.gateway | json }}
+                  amount: "0.0"
+                  gateway: {{ transaction.gateway | json }}
+                  kind: VOID
+                  orderId: {{ order.id | json }}
+                  parentId: {{ transaction.id | json }}
                 }
-              ],
+              ]
             {% endif %}
-            "refund_line_items": [
-              {% for line_item in order.line_items %}
+            refundLineItems: [
+              {% assign line_items = order.lineItems.edges | map: "node" %}
+              {% for line_item in line_items %}
                 {
-                  "line_item_id": {{ line_item.id | json }},
-
-                  {% if options.cancel_fulfillment_and_restock_to_default_location__boolean and line_item.variant_inventory_management == "shopify" %}
-                    "restock_type": "cancel",
-                    "quantity": {{ line_item.fulfillable_quantity | default: line_item.quantity | json }},
-                    "location_id": {{ shop.primary_location_id | json }}
+                  lineItemId: {{ line_item.id | json }}
+                  {% if options.restock_line_items__boolean and line_item.variant.inventoryItem.tracked %}
```

Comment:
Looking through the schema, I think we can do better than my original suggestion here: I just found [LineItem.restockable](https://shopify.dev/docs/admin-api/graphql/reference/orders/lineitem#restockable-2020-10)! Query for this from the suggested RefundLineItem, as in `lineItem { restockable }`.